src/vision/status_quo.md

```diff
@@ -123,22 +123,31 @@ Outline:
+Soon enough, she and her team have a proof of concept working. They run some local stress tests and notice that 5% of requests hang and fail to respond. The client eventually times out. She cannot reproduce this problem when running one-off requests locally. It only shows up when sending above 200 requests-per-second. 
 
-She realizes that she doesn't have any tooling that can give her insight into what's going on. She starts to add lots of logging, attempting to tie log entries to specific connections. Using an operating system tool, she is able to identify the socket addresses for the hung connections, so she also includes the socket addresses in each log message. She can then grep the logs to find those that are associated with the hung connections. Of course, the logs only tell her what the connection managed to do successfully, they don't tell her why it stopped -- so she has to keep going back and adding more logs until she can narrow down the exact call that hangs.
+She realizes that she doesn't have any tooling to give her insight into what's going on. She starts to add lots of logging, attempting to tie log entries to specific connections. Using an operating system tool, she can identify the socket addresses for the hung connections, so she also includes the socket addresses in each log message. She then filters the logs to find entries associated with hung connections. Of course, the logs only tell her what the connection managed to do successfully; they don't tell her why it stopped -- so she keeps going back to add more logging until she can narrow down the exact call that hangs.
 
-Eventually, she identifies that the last log message is right before calling the HMAC authentication library that is written C, and is synchronous, but only when refreshing the private authentication token from the source of truth. 
+Eventually, she identifies that the last log message is right before authenticating the request. An existing C library performs authentication, integrated with the routing service using a custom future implementation. She eventually finds a bug in the implementation that resulted in occasional lost wake-ups.
+
+She fixes the bug. The service is now working as expected and meeting Grace's performance goals.
 
 #### 🤔 Frequently Asked Questions
 
 * **What are the morals of the story?**
-    * (Explain your key points here)
-* **Are you just making this stuff up?**
-    * (Cite important sources here; feel free to add files into the conversations folder for more details)
+    * When coming from a background of network engineering, users will bring their own design choices around architecture.
+        * Examples: [seastar](http://seastar.io/) and [Glommio](https://www.datadoghq.com/blog/engineering/introducing-glommio/)
+    * There is a lack of debugging tools for async.
+    * Writing futures by hand is error prone.
+* **What are the sources for this story??**
```

Comment:
Two question marks here—is this intended? Added by 9e18afe0bd72a90f712bf46e946607d9631b5975.

Reply:
Not intended :)